semgrep-core/src/parsing/tree_sitter/Parse_swift_tree_sitter.ml

```diff
@@ -2714,7 +2714,19 @@ and map_tuple_expression (env : env)
       v4
   in
   let v5 = (* ")" *) token env v5 in
-  G.Container (G.Tuple, (v1, v3 :: v4, v5)) |> G.e
+  let exprs = v3 :: v4 in
+  match exprs with
+  (* The grammar doesn't differentiate, but according to the Swift spec:
+   *
+   * "A single expression inside parentheses is a parenthesized expression."
+   *
+   * https://docs.swift.org/swift-book/ReferenceManual/Expressions.html#ID552
+   *
+   * See also
+   * https://docs.swift.org/swift-book/ReferenceManual/Expressions.html#ID395
+   *)
+  | [ e ] -> e
```

Comment:
actually we have also now ParenExpr that is useful for autofix, but it's not used everywhere.

Reply:
I saw that, but the comment said it was only needed in rare cases and I checked the JS behavior and it's not used there.